ilastik/utility/slot_name_enum.py

```diff
@@ -0,0 +1,53 @@
+import enum
+from typing import List, Tuple
+from itertools import chain
+
+class SlotNameEnum(enum.IntEnum):
+    """A map from slot names to their indices within a multislot
+
+    Do note that the ENUM_NAMEs that you pick matter, as they determine
+    the human readable version that you get from instances of this enum:
+
+    e.g a value defined like so:
+        MY_DATA_SLOT = 123
+    will have its slotName rendered as
+        'My Data Slot'
+    """
+
+    @property
+    def slotName(self) -> str:
```

Comment:
I would also add that implicitly deriving GUI labels from constant names is probably not a good idea, maybe it would be better to explicitly specify the "display name"? Something like `MY_PRECIOUS_SLOT = SlotDef(42, "My Precious Slot")`.

Reply:
There is so much wrong with these enums that there is almost no point investing more time into making it "better", as it is already fundamentally wrong. They are just a less messy implementation of current logic.

What we should eventually do is have slots be defined in the data selection operator  as proper slots, so that they can even be marked as `optional` instead of having them be subslots in the generic Input slot =/